profile-widget/profilewidget2.cpp

```diff
@@ -2103,28 +2103,40 @@ bool ProfileWidget2::PictureEntry::operator< (const PictureEntry &e) const
 
 // Calculate the y-coordinates of the thumbnails, which are supposed to be sorted by x-coordinate.
 // This will also change the order in which the thumbnails are painted, to avoid weird effects,
-// when items are added later to the scene. This is simply done by increasing the Z-value.
+// when items are added later to the scene. This is done using the QGraphicsItem::packBefore() function.
+// We can't use the z-value, because that will be modified on hoverEnter and hoverExit events.
 void ProfileWidget2::calculatePictureYPositions()
 {
+	// Quit early if there are no items. The last loop in this function assumes that the vector is not empty.
+	if (pictures.empty())
+		return;
+
 	double lastX = -1.0, lastY;
```

Comment:
@bstoeger 
Already with the old code but still with the new code there is a warning about lastY:
/home/stefan/Entwicklung/Subsurface/subsurface/profile-widget/profilewidget2.cpp:2123:17: warning: 'lastY' may be used uninitialized in this function [-Wmaybe-uninitialized]
    y = lastY + 3;
Maybe you could silence this with this change?

Reply:
yes, please initialize it to `0.0`.

Reply:
Ok - stupid compiler. :(